core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1113,8 +1113,8 @@ class KafkaController(val config: KafkaConfig,
     // so we will keep the previous behavior and don't reject the request
     if (brokerEpoch != AbstractControlRequest.UNKNOWN_BROKER_EPOCH) {
       val cachedBrokerEpoch = controllerContext.liveBrokerIdAndEpochs(id)
-      if (brokerEpoch < cachedBrokerEpoch) {
-        val stateBrokerEpochErrorMessage = "Received controlled shutdown request from an old broker epoch " +
+      if (brokerEpoch != cachedBrokerEpoch) {
+        val stateBrokerEpochErrorMessage = "Received controlled shutdown request from an invalid broker epoch " +
```

Comment:
Probably a good idea to mention the motivation for this in the PR description.

Reply:
I may revert this change. Just need to remember why I added it...